main/src/cgeo/geocaching/ui/CacheListAdapter.java

```diff
@@ -428,23 +426,14 @@ public View getView(final int position, final View rowView, final ViewGroup pare
         compasses.add(holder.direction);
         holder.direction.setTargetCoords(cache.getCoords());
 
-        Spannable spannable = null;
         if (cache.isDisabled() || cache.isArchived() || CalendarUtils.isPastEvent(cache)) { // strike
-            spannable = Spannable.Factory.getInstance().newSpannable(cache.getName());
-            spannable.setSpan(new StrikethroughSpan(), 0, spannable.toString().length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+            holder.text.setPaintFlags(holder.text.getPaintFlags() | Paint.STRIKE_THRU_TEXT_FLAG);
```

Comment:
Does this really work in a list containing both kinds of caches? I thought that with the view holder pattern, we always must make sure all relevant flags are set for each element of the list. Therefore I would expect that we need to set the paint flags and the text color to a common standard every time before overriding one of them in the if statements. Otherwise the we might end up e.g. with an old paint flag from one of the previous iterations, even though we now want to render a non-disabled cache.